### ISSUE TITLE
Local variable escaped from the function

### ISSUE THREAD

**Issue by A4-Tacks** [koalaman/shellcheck#3044]
#### For bugs
- Rule Id (if any, e.g. SC1000):
- My shellcheck version (`shellcheck --version` or "online"):
  ```bash
  ShellCheck - shell script analysis tool
  version: 0.9.0
  license: GNU General Public License, version 3
  website: https://www.shellcheck.net
  ```
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
f() {
    local x=
}
echo "$x"
```

#### Here's what shellcheck currently says:
```bash
```

#### Here's what I wanted or expected to see:
```bash

In - line 5:
echo "$x"
      ^-- SC2154 (warning): x is referenced but not assigned.

For more information:
  https://www.shellcheck.net/wiki/SC2154 -- x is referenced but not assigned.
```
